src/battle_transition.c

```diff
@@ -0,0 +1,2513 @@
+#include "global.h"
+#include "battle_transition.h"
+#include "main.h"
+#include "rom4.h"
+#include "task.h"
+#include "palette.h"
+#include "trig.h"
+#include "field_effect.h"
+#include "rng.h"
+#include "sprite.h"
+#include "sound.h"
+#include "songs.h"
+#include "trainer.h"
+#include "field_camera.h"
+
+void sub_807DE10(void);
+void dp12_8087EA4(void);
+
+extern u8 ewram[];
+extern u16 gUnknown_03005560[];
+extern u16 gUnknown_03004DE0[][0x3C0];
+
+extern const struct OamData gFieldOamData_32x32;
+
+struct TransitionData
+{
+    vs8 VBlank_DMA;
+    u16 WININ;
+    u16 WINOUT;
+    u16 field_6;
+    u16 WIN0V;
+    u16 field_A;
+    u16 field_C;
+    u16 BLDCNT;
+    u16 BLDALPHA;
+    u16 BLDY;
+    s16 field_14;
+    s16 field_16;
+    s16 field_18;
+    s16 field_1A;
+    s16 field_1C;
+    s16 field_1E; // unused
+    s16 field_20;
+    s16 field_22; // unused
+    s16 data[11];
```

Comment:
Only Task uses an array. `TRANSITION_STRUCT.data.bar` is easier to read than `TRANSITION_STRUCT.data[4]` or `TRANSITION_STRUCT.data[BAR]`.